radix-engine-interface/src/api/event_api.rs

```diff
@@ -1,14 +1,16 @@
 use crate::api::types::*;
-use crate::blueprints::resource::*;
+use crate::blueprints::resource::Resource;
 
 // TODO: more thinking on whether should be part of the ClientApi.
-pub trait ClientMeteringApi<E> {
-    fn lock_fee(
+pub trait ClientEventApi<E> {
+    fn consume_cost_units(&mut self, units: u32) -> Result<(), E>;
+
+    fn credit_cost_units(
         &mut self,
         vault_id: VaultId,
-        fee: Resource,
+        locked_fee: Resource,
         contingent: bool,
     ) -> Result<Resource, E>;
 
-    fn consume_cost_units(&mut self, units: u32) -> Result<(), E>;
+    fn update_instruction_index(&mut self, new_index: usize) -> Result<(), E>;
```

Comment:
Seems like a strange api? Is this temporary?

Reply:
It is strange. 

In my follow-up PR, I rebranded it as `ClientUnsafeApi` (which enables "trusted" clients to interact with kernel modules).

Reply:
hrm, not sure about that abstraction, still seems a bit weird but we can discuss when we get to it.

Reply:
sure